legacy/src/main/java/org/opensearch/sql/legacy/cursor/DefaultCursor.java

```diff
@@ -132,8 +220,18 @@ private JSONObject schemaEntry(String name, String alias, String type) {
     return entry;
   }
 
-  private static String encodeCursor(JSONObject cursorJson) {
-    return Base64.getEncoder().encodeToString(cursorJson.toString().getBytes());
+  @SneakyThrows
+  private static String encodeCursor(JSONObject cursorJson, SearchSourceBuilder sourceBuilder) {
+    String jsonBase64 = Base64.getEncoder().encodeToString(cursorJson.toString().getBytes());
+
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    XContentBuilder builder = XContentFactory.jsonBuilder(outputStream);
+    sourceBuilder.toXContent(builder, null);
+    builder.close();
+
+    String searchRequestBase64 = Base64.getEncoder().encodeToString(outputStream.toByteArray());
+
+    return jsonBase64 + ":::" + searchRequestBase64;
```

Comment:
Also,, can we add searchSourceBuilder as a filed in json instead of adding it to cursor.
```
 XContentBuilder builder = XContentFactory.jsonBuilder();
        sourceBuilder.toXContent(builder, null);
        String searchRequestJson = builder.string();
```
```
String searchRequestJson = new String(searchRequestBytes);
        
        XContentParser parser = XContentType.JSON.xContent().createParser(
            xContentRegistry, 
            IGNORE_DEPRECATIONS, 
            searchRequestJson
        );
        
        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.fromXContent(parser);
        ```


Add searchRequestJson to JSON

 @SneakyThrows is not good in source code. Catch the exception and log the errors.